records_mover/records/delimited/sniff.py

```diff
@@ -1,143 +1,297 @@
 import chardet
-from .types import RecordsHints, BootstrappingRecordsHints, HintEncoding
-from .csv_streamer import stream_csv
-from .conversions import python_encoding_from_hint
+from contextlib import contextmanager
+from . import RecordsHints, BootstrappingRecordsHints
+from .csv_streamer import stream_csv, python_encoding_from_hint
 import io
 import logging
-from .types import MutableRecordsHints
-from .conversions import (
-    hint_compression_from_pandas,
-    hint_encoding_from_pandas,
-    hint_encoding_from_chardet
-)
-from typing import List, IO, Optional, TYPE_CHECKING
-if TYPE_CHECKING:
-    from pandas.io.parsers import TextFileReader
+import csv
+import gzip
+import bz2
+from .types import HintEncoding, HintRecordTerminator, HintQuoting, HintCompression
+from .conversions import hint_encoding_from_chardet
+import pandas
+from typing import List, IO, Optional, Iterator, NoReturn, Dict
 
 
 logger = logging.getLogger(__name__)
 
+HINT_INFERENCE_SAMPLING_SIZE_BYTES = 1024
```

Comment:
Let me know if you think this magic number feels important enough to add in as a ProcessingInstruction.  My gut is that the default value is small enough not to matter performance-wise and big enough that we're not likely to get better information out of the hint sniffers.

This could potentially be something we offer config via the mechanism in https://github.com/bluelabsio/records-mover/pull/47 - though if it does make sense as a parameter, I'm not sure it makes sense as a system-wide parameter...